SmartDeviceLinkTests/ProxySpecs/SDLHapticManagerSpec.m

```diff
@@ -0,0 +1,169 @@
+//
+//  SDLHapticManagerSpec.m
+//  SmartDeviceLink-iOS
+//
+//  Copyright © 2017 smartdevicelink. All rights reserved.
+//
+
+#import <Quick/Quick.h>
+#import <Nimble/Nimble.h>
+#import <OCMock/OCMock.h>
+
+#import "SDLHapticManager.h"
+#import "SDLSendHapticData.h"
+#import "SDLManager.h"
+#import "SDLTouchCoord.h"
+#import "SDLTouchEvent.h"
+#import "SDLTouch.h"
+
+SDLTouch* generateTouchEvent(int xCoord, int yCoord)
+{
+    unsigned long timeStamp = 10;
+    SDLTouchCoord* firstCoord = [[SDLTouchCoord alloc] init];
+    firstCoord.x = [NSNumber numberWithInt:xCoord];
+    firstCoord.y = [NSNumber numberWithInt:yCoord];
+    
+    SDLTouchEvent* firstTouchEvent = [[SDLTouchEvent alloc] init];
+    firstTouchEvent.touchEventId = @0;
+    firstTouchEvent.coord = [NSMutableArray arrayWithObject:firstCoord];
+    firstTouchEvent.timeStamp = [NSMutableArray arrayWithObject:@(timeStamp)];
+    SDLTouch* firstTouch = [[SDLTouch alloc] initWithTouchEvent:firstTouchEvent];
+    return firstTouch;
+}
+
+QuickSpecBegin(SDLHapticManagerSpec)
+
+describe(@"the haptic manager", ^{
+    
+    __block UIWindow *uiWindow;
+    __block UIViewController *uiViewController;
+    
+    __block SDLHapticManager *hapticManager;
+    __block SDLSendHapticData* capturedHapticRPC;
+    
+    __block id sdlManagerMock = OCMClassMock([SDLManager class]);
+    
+    beforeEach(^{
```

Comment:
Reset the `hapticManager` to nil if it's reset in every context. Reset the `capturedHapticRPC` to nil here as well.